android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/survey/SurveyScreen.kt

```diff
@@ -44,41 +64,58 @@ fun SurveyScreen(
 @OptIn(ExperimentalPagerApi::class)
 @Composable
 private fun SurveyScreenContent(
-    questions: List<QuestionUiModel>,
+    scaffoldState: ScaffoldState,
+    isLoading: Boolean,
+    survey: SurveyUiModel?,
     onBackClick: () -> Unit,
     onSubmitClick: () -> Unit
 ) {
     val pagerState = rememberPagerState()
     val scope = rememberCoroutineScope()
 
-    HorizontalPager(
-        count = questions.size + 1,
-        state = pagerState,
-        userScrollEnabled = false,
-        modifier = Modifier.fillMaxSize()
-    ) { index ->
-        // TODO use question.displayType instead
-        if (index == 0) {
-            SurveyIntro(
-                onBackClick = onBackClick,
-                onStartClick = { pagerState.scrollToNextPage(scope) }
-            )
-        } else {
-            SurveyQuestion(
-                index = index,
+    Scaffold(scaffoldState = scaffoldState) { padding ->
+        survey?.let {
+            val questions = survey.questions
+            HorizontalPager(
                 count = questions.size,
-                question = questions[index - 1],
-                onCloseClick = onBackClick,
-                onNextClick = {
-                    if (index != questions.size) {
-                        pagerState.scrollToNextPage(scope)
-                    } else {
-                        onSubmitClick()
-                    }
+                state = pagerState,
+                userScrollEnabled = false,
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(padding)
+            ) { index ->
+                // TODO use question.displayType instead
+                if (index == 0) {
+                    SurveyIntro(
+                        survey = survey,
+                        onBackClick = onBackClick,
+                        onStartClick = { pagerState.scrollToNextPage(scope) }
+                    )
+                } else {
+                    SurveyQuestion(
+                        index = index,
+                        count = questions.size - 1,
+                        question = questions[index],
+                        onCloseClick = onBackClick,
+                        onNextClick = {
+                            if (index != questions.size) {
+                                pagerState.scrollToNextPage(scope)
+                            } else {
+                                onSubmitClick()
+                            }
```

Comment:
will there be viewModel for this logic 🤔

Reply:
@blyscuit Why do we need to include this UI interaction in ViewModel?

Reply:
Separation of concern and reuse for other platform. No problem with this as well. 🙌